routes/index.js

```diff
@@ -44,96 +47,70 @@ router.get('/test-mail', function (req, res) {
 // Route for login/adding new user
 // If request is missing parameters, returns 400
 // If request is for existing user, returns 409
-// Else, creates user and returns 201 along with user data
-router.post('/api/users', function (req, res) {
+// Else, creates user and returns 201 along with user secret
+router.post('/api/users', async function (req, res) {
   const { firstname, lastname, email } = req.body
   const secret = uuidv4()
 
   if (!firstname || !lastname || !email) {
     console.log('Input field empty')
     return res.status(400).send('Input field empty')
   }
-  // Checks for duplicate
-  emailAlreadyRegistered(email)
-    .then(result => {
-      if (result) {
-        res.status(409).send({ message: 'exist' })
-      } else {
-        const insertUser = {
-          text: 'INSERT INTO users(firstname, lastname, email, secret) VALUES ($1, $2, $3, $4)',
-          values: [firstname.toLowerCase(), lastname.toLowerCase(), email.toLowerCase(), secret]
-        }
-        db.query(insertUser)
-          .then(result => {
-            res.send({ secret: secret })
-          })
-          .catch(err => {
-            res.status(500).json(err)
-          })
-      }
-    })
-})
 
-// Helper function for checking user duplicates using email
-// Goes through database and counts where email matches
-async function emailAlreadyRegistered (email) {
-  const checkdup = {
-    text: 'SELECT COUNT(email) FROM users WHERE email = $1',
-    values: [email.toLowerCase()]
+  const result = await que.emailAlreadyRegistered(email)
+  if (result) {
+    res.status(409).send()
+  } else {
+    que.addUser(firstname, lastname, email, secret)
+      .then(() => {
+        res.send({ secret: secret })
+      })
+      .catch(err => {
+        res.status(500).json(err)
+      })
   }
-  const result = await db.query(checkdup)
-  return(result.rows[0].count > 0) 
-}
+})
 
 // Route for getting user data from secret
 router.get('/api/user/:secret', function (req, res) {
-  const getUserData = {
-    text: 'SELECT users.firstname, users.lastname, u.firstname AS friendfname, u.lastname As friendlname ' +
-            'FROM users ' +
-            'LEFT JOIN friendships AS f ON  f.user1 = users.id ' +
-            'LEFT JOIN users as u ON f.user2 = u.id WHERE users.secret = $1',
-    values: [req.params.secret]
-  }
-  db.query(getUserData)
+  que.getUserData(req.params.secret)
     .then(results => {
-      res.send(formatUserData(results))
+      res.send(pro.formatUserData(results))
     })
     .catch(err => {
       res.status(404).json(err)
     })
 })
 
-// Helper function for /api/user/:secret
-// Takes in res data and formats it to user first name, user last name, friends list
-function formatUserData (res) {
-  const userData = {}
-  const friends = res.rows
-  const flist = []
-  if (friends[0].friendfname == null) {
-    flist[0] = 'You currently have no friends'
+router.post('/api/friendships', async function (req, res) {
```

Comment:
Looks great -- much cleaner code, now that you've split out the DB queries and used await more.